### PR TITLE
nan to -inf

### PR DESCRIPTION
**Pull request by danjust**
Another tiny fix: Currently, diverging models that return nan get perfect evaluation (since `candidate_score>torch.nan = False`). The alternative to this fix would be to compute `n_worse = torch.sum(candidate_score <= pos_score, dim=-1)`

### CODE REVIEW

**Review by AlCatt91 — APPROVED** [review]
Thanks, Daniel, good point! It might be slightly more efficient (?) computing the `n_worse` as you suggested, but I don't expect this `nan_to_num_` replacement to add too much overhead to the metric computation, so I think it's also fine to leave it like this.
